### PR TITLE
https://github.com/JosephSilber/bouncer/issues/245

### PR DESCRIPTION
**Pull request by ecodrutz**

### CODE REVIEW

**Comment by ecodrutz** [comment]
Hello,
It seems Travis had a internet problem when it ran the tests and two failed with timeouts to the composer and github websites.
Maybe you can rerun them to ensure there are no errors.

File: src/Bouncer.php [hunk 1]
```diff
@@ -248,10 +248,6 @@ public function gate()
             return $this->gate;
         }
 
-        if ($throw) {
-            throw new RuntimeException('The gate instance has not been set.');
-        }
-
```
Comment (JosephSilber) [inline src/Bouncer.php]: Why did you remove this?

File: src/Bouncer.php [hunk 2]
```diff
@@ -373,52 +369,44 @@ public function ownedVia($model, $attribute = null)
...
      * @param  string  $model
-     * @return $this
+     * @return void
      */
     public static function useAbilityModel($model)
     {
         Models::setAbilitiesModel($model);
-
-        return $this;
     }
```
Comment (JosephSilber) [inline src/Bouncer.php]: What's wrong here is that they were static in the first place, not that they returned `$this`.

File: tests/BaseTestCase.php [hunk 3]
```diff
@@ -46,7 +46,7 @@
      */
     public function setUp()
     {
-        Models::setUsersModel(User::class);
+        Bouncer::useUserModel(User::class);
```
Comment (JosephSilber) [inline tests/BaseTestCase.php]: Why did you change this?

Also, since those methods on the Bouncer class will now be non-static, we can't even use `Bouncer` here.

File: src/Bouncer.php [hunk 4]
```diff
@@ -248,10 +248,6 @@ public function gate()
             return $this->gate;
         }
 
-        if ($throw) {
-            throw new RuntimeException('The gate instance has not been set.');
```
Comment (JosephSilber) [inline src/Bouncer.php]: Why did you remove this throw?

**Comment by JosephSilber** [comment]
Fixed in https://github.com/JosephSilber/bouncer/commit/1a774ea3913c14d26d8b12d7194bec22a740afd8

File: tests/BaseTestCase.php (hunk 3 above)
Comment (ecodrutz) [reply tests/BaseTestCase.php]: Hello,
I changed this because there is the method on the Bouncer class that wraps up the code, and to make sure the tests fail in case there is an error like it was with returning $this in a static context.
As you stated the error was that the method shouldn't be static, not that it returned $this, so now I understand it, and it`s not correct in the new context.

File: src/Bouncer.php (hunk 4 above)
Comment (ecodrutz) [reply src/Bouncer.php]: Hello,
This throw will never happen because a few commits back the variable was removed from the method, and is always null:
    /**
     * Get the gate instance. Throw if not set.
     *
     * @return \Illuminate\Contracts\Auth\Access\Gate
     *
     * @throws \RuntimeException
     */
    public function gate()
    {
        if ($this->gate) {
            return $this->gate;
        }
        return null;
    }
this is how the method looks now on the beta5, and it has no variable throw received as parameter.

Comment (JosephSilber) [reply src/Bouncer.php]: I understand why you removed the conditional. That was wrong :+1: 

I was asking why you removed the throw. I changed it here: https://github.com/JosephSilber/bouncer/commit/485dbe0007171e284e5ae99f1480c6bb6e54036e

Comment (ecodrutz) [reply src/Bouncer.php]: I thought since it was never actually thrown, it wasn`t needed, so I removed it completely. I guess it will never return null now as it used to, correct?

Comment (JosephSilber) [reply src/Bouncer.php]: This method is used internally, where certain methods in the `Bouncer` class need to call methods directly on the `Gate`. Instead of checking it each time, we check it here once.
